scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -418,6 +419,17 @@ class FossId internal constructor(
     ): Pair<String, String> {
         val projectRevision = context.labels[PROJECT_REVISION_LABEL]
 
+        val vcs = requireNotNull(VersionControlSystem.forUrl(url))
```

Comment:
IIRC, there is some URL-mapping support for the repository URLs. Does this need to be taken into account here?

Reply:
Also, the URL provider kicks in just before creating the scan.

It is mainly used for two things: add credentials, rewrite some part of the URL (e.g. changing the scheme).

For the credentials, this does not need to be done for the `VersionControlSystem` as it comes to it's own delegation to the Authenticator:
see https://github.com/oss-review-toolkit/ort/blob/c4a7a749ad214137bedfd7410c873596abb614f6/downloader/src/main/kotlin/vcs/Git.kt#L70.

However for the rewriting of the URL, I am not sure anymore. If you change the scheme, it could be because the given scheme is unsupported.

But how would you solve this issue ? We cannot call the URL provider because we don't want the credentials being added to the URL (see above).

What is your opinion ?

Reply:
I retested the changes with a pipeline using the URL scheme rewriting and it ran without error.